### ISSUE TITLE
Support Swift tools version 5.3.0

### ISSUE THREAD

**Issue by achirkof** [mattpolzin/swift-codecov-action#1]
Now it's only possible to build using Swift tools version 5.2.0. Do you plan to add Swift tools version 5.3.0?

**Comment by mattpolzin** [comment]
I am guessing you are referring to building the https://github.com/mattpolzin/swift-test-codecov tool on top of which the swift-codecov-action is built because the action itself (this repository) should not depend on Swift at all.

As for swift-test-codecov (the tool that powers this swift-codecov-action), I do plan to make it build under Swift 5.3, although I have not tried building it with Swift 5.3 yet.

**Comment by achirkof** [comment]
I have an SPM package that can be built only with Swift tools version 5.3.0 which comes from Xcode 12. This is because the package contains assets. 
And yes, it's relevant to another repo you mentioned where you use swift-tools-version:5.1 if I'm right.

**Comment by mattpolzin** [comment]
So if I understand correctly, you are using https://github.com/mattpolzin/swift-test-codecov as a dependency in your Swift 5.3 package and you are hitting a build error. I have not heard any indication that Swift tools version 5.3 would be unable to resolve a package that uses Swift tools version 5.1, but it's possible that there's a bug they will fix prior to release. What error are you getting (and am I correct that you are using my package as a dependency in your project, or are you trying to build my package directly using Swift 5.3)?

**Comment by achirkof** [comment]
I don't use your package as a dependency, I use your action for Github and get the error:

<img width="1339" alt="Screenshot 2020-07-02 at 21 14 05" src="https://user-images.githubusercontent.com/1292136/86395615-3604ca80-bca9-11ea-801c-61dca47e77e4.png">

Here is my workflow config [ci.yml](https://github.com/achirkof/LoadableImage/actions/runs/155292921/workflow)

**Additional**: to use your action I have to use `runs-on: ubuntu-latest`, but my package can't be built on ubuntu. I think that's the source of the problem.

**Comment by mattpolzin** [comment]
Oh, I see what's going on now. I was imaging a different failure. Thanks, the screenshot and link to your workflow file are very helpful.

The problem can actually be solved in your workflow config; You can specify a different container to run the code coverage job within. In your case, you'll need to pick a container that has Swift 5.3 installed. I recommend `swiftlang:nightly-5.3` for now and once Swift 5.3 is officially released you can switch to using `swift:5.3`.

Modify your job as follows (just the part about a `container` before `runs-on` is new):
```yaml
  Test:
    name: Tests
    container:
      image: swiftlang:nightly-5.3
    runs-on: ubuntu-latest
    steps:
    ...
```

**Comment by achirkof** [comment]
> Oh, I see what's going on now. I was imaging a different failure. Thanks, the screenshot and link to your workflow file are very helpful.
> 
> The problem can actually be solved in your workflow config; You can specify a different container to run the code coverage job within. In your case, you'll need to pick a container that has Swift 5.3 installed. I recommend `swiftlang:nightly-5.3` for now and once Swift 5.3 is officially released you can switch to using `swift:5.3`.
> 
> Modify your job as follows (just the part about a `container` before `runs-on` is new):
> 
> ```yaml
>   Test:
>     name: Tests
>     container:
>       image: swiftlang:nightly-5.3
>     runs-on: ubuntu-latest
>     steps:
>     ...
> ```

Looks like this image is private

<img width="1259" alt="Screenshot 2020-07-02 at 21 35 47" src="https://user-images.githubusercontent.com/1292136/86397224-1b802080-bcac-11ea-9395-4422b2be11dd.png">

**Comment by mattpolzin** [comment]
My apologies, should have been `swiftlang/swift:nightly-5.3`.

The full list of tags for the `swiftlang/swift` repository can be found here: https://hub.docker.com/r/swiftlang/swift/tags

**Comment by achirkof** [comment]
> My apologies, should have been `swiftlang/swift:nightly-5.3`.
> 
> The full list of tags for the `swiftlang/swift` repository can be found here: https://hub.docker.com/r/swiftlang/swift/tags

It's now complaining on SwiftUI, but locally `swift test --enable-test-discovery --enable-code-coverage` executes with success.

<img width="1155" alt="Screenshot 2020-07-02 at 22 35 51" src="https://user-images.githubusercontent.com/1292136/86402529-a7964600-bcb4-11ea-9e07-b36cd6325025.png">

**Comment by mattpolzin** [comment]
Hmmm, yeah, I've not run into this before but because GitHub cannot run Docker on Mac OS and your project cannot be tested on Linux (because SwiftUI is not available there) the current flow for this action will not work. If there is a way to collect test coverage for your project in a Mac-based GitHub Action and then pass that context to my Action running in a Linux container then you could still use this action, but I don't have the time to look into whether that is possible or not right now.

The TLDR; is that unless you are interested in digging into how GitHub Actions work more deeply, this action is not going to be a viable solution for your project.

**Comment by achirkof** [comment]
I'm new to Github Actions. I used Travis before. I'll try to find a solution. Thank you for your help! 🤝

**Comment by mattpolzin** [comment]
You’re welcome. 

I’m going to close this for now. 

It occurs to me that the tool underlying this script (not strictly speaking this action, which relies on a Docker container) _could_ be built and run on a GitHub Mac machine. Still, this needs a bit of exploration before I could really explain how to do it in a README.
